kubectl-minio/cmd/tenant-delete.go

```diff
@@ -103,41 +103,42 @@ func (d *tenantDeleteCmd) run(args []string) error {
 	if err != nil {
 		return err
 	}
-	return deleteTenant(oclient, kclient, d, args[0])
+	for _, arg := range args {
+		if err = deleteTenant(oclient, kclient, d, arg); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func deleteTenant(client *operatorv1.Clientset, kclient *kubernetes.Clientset, d *tenantDeleteCmd, name string) error {
 	tenant, err := client.MinioV2().Tenants(d.ns).Get(context.Background(), name, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
+
 	if err := client.MinioV2().Tenants(d.ns).Delete(context.Background(), name, v1.DeleteOptions{}); err != nil {
 		return err
 	}
-	if err := kclient.CoreV1().Secrets(d.ns).Delete(context.Background(), tenant.Spec.CredsSecret.Name, metav1.DeleteOptions{}); err != nil {
-		return err
-	}
-	if len(tenant.Spec.Users) > 0 {
-		if err := kclient.CoreV1().Secrets(d.ns).Delete(context.Background(), tenant.Spec.Users[0].Name, metav1.DeleteOptions{}); err != nil {
-			return err
-		}
-	}
 
-	fmt.Printf("Deleting MinIO Tenant %s\n", tenant.ObjectMeta.Name)
-	fmt.Printf("Deleting MinIO Tenant Credentials Secret %s\n", tenant.Spec.CredsSecret.Name)
+	fmt.Printf("Deleting MinIO Tenant: %s\n", name)
+
+	// Delete credentials secret, ignore any errors.
+	kclient.CoreV1().Secrets(d.ns).Delete(context.Background(), tenant.Spec.CredsSecret.Name,
```

Comment:
are we sure? loosing the root credentials could lead to permantent data loss o_o

Reply:
what are we losing @dvaldivia ? - root credentials are not used for anything other than accessing the API 

Also, this is a delete call there is a confirmation from the user.  This is not different than what we have. 

The main reason to not return errors is that it's bogus to return errors in this situation - deletion is a permanent loss of a tenant anyways. 

The command doesn't support any rollback so I don't see what is the point of returning errors.

Another thing to notice is that the command itself is broken throws weird errors when deleting users, leaves resources hanging which requires manual cleanup.